nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -344,39 +301,42 @@ public DocumentModel createComment(DocumentModel docModel, DocumentModel parent,
 
     @Override
     public List<DocumentModel> getDocumentsForComment(DocumentModel comment) {
-        Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(ResourceAdapter.CORE_SESSION_CONTEXT_KEY,
-                comment.getCoreSession());
-        RelationManager relationManager = Framework.getService(RelationManager.class);
-        Graph graph = relationManager.getGraph(config.graphName, comment.getCoreSession());
-        Resource commentResource = relationManager.getResource(config.commentNamespace, comment, ctxMap);
-        if (commentResource == null) {
-            throw new NuxeoException("Could not adapt document model to relation resource ; "
-                    + "check the service relation adapters configuration");
-        }
-        Resource predicate = new ResourceImpl(config.predicateNamespace);
-
-        List<Statement> statementList = graph.getStatements(commentResource, predicate, null);
-        if (graph instanceof JenaGraph) {
-            // XXX AT: BBB for when repository name was not included in the
-            // resource uri
-            Resource oldDocResource = new QNameResourceImpl(config.commentNamespace, comment.getId());
-            statementList.addAll(graph.getStatements(oldDocResource, predicate, null));
-        }
-
-        List<DocumentModel> docList = new ArrayList<DocumentModel>();
-        for (Statement stmt : statementList) {
-            QNameResourceImpl subject = (QNameResourceImpl) stmt.getObject();
-            DocumentModel docModel = (DocumentModel) relationManager.getResourceRepresentation(config.documentNamespace,
-                    subject, ctxMap);
-            if (docModel == null) {
-                log.warn("Could not adapt comment relation subject to a document "
-                        + "model; check the service relation adapters configuration");
-                continue;
+        return CoreInstance.doPrivileged(comment.getRepositoryName(), session -> {
+            Map<String, Object> ctxMap = Collections.<String, Object> singletonMap(
```

Comment:
`Collections.<String, Object> singletonMap` can be replaced directly by `Collections.singletonMap`

Reply:
This one is wraping old code with CoreInstance#doPrivilledged.
Do I clean it or just do the bare minimum ?
wdyt @troger